src/utilities/flattenParentMatches.ts

```diff
@@ -0,0 +1,9 @@
+import { Resolved, Route, isNamedRoute, isPublicRoute } from '@/types'
+
+export function flattenParentMatches(routes: Resolved<Route>): string {
+  const namedAndPublicRoutesNames = routes.matches
+    .filter(route => isNamedRoute(route) && isPublicRoute(route))
```

Comment:
same question applies here, not sure we need name and public, possible we're not applying that business rule correctly throughout

Reply:
Yeah why'd you do that lol. Removed